crates/matrix-sdk-appservice/src/error.rs

```diff
@@ -41,43 +41,49 @@ pub enum Error {
     #[error("uri path is unknown")]
     UriPathUnknown,
 
-    #[error(transparent)]
-    HttpRequest(#[from] ruma::api::error::FromHttpRequestError),
+    #[error("rum api from http request error: {0}")]
+    FromHttpRequest(#[from] ruma::api::error::FromHttpRequestError),
 
-    #[error(transparent)]
+    #[error("ruma id parser error: {0}")]
     Identifier(#[from] ruma::IdParseError),
 
-    #[error(transparent)]
+    #[error("http error: {0}")]
     Http(#[from] http::Error),
 
-    #[error(transparent)]
+    #[error("sdk http error: {0}")]
+    SdkHttp(#[from] matrix_sdk::HttpError),
+
+    #[error("sdk store error: {0}")]
+    SdkStore(#[from] matrix_sdk::StoreError),
+
+    #[error("url parse error: {0}")]
     Url(#[from] url::ParseError),
 
-    #[error(transparent)]
+    #[error("serde de error: {0}")]
     Serde(#[from] serde::de::value::Error),
 
-    #[error(transparent)]
+    #[error("io error: {0}")]
     Io(#[from] std::io::Error),
 
-    #[error(transparent)]
+    #[error("http uri invalid error: {0}")]
     InvalidUri(#[from] http::uri::InvalidUri),
 
-    #[error(transparent)]
+    #[error("sdk error: {0}")]
     Matrix(#[from] matrix_sdk::Error),
```

Comment:
Maybe this one should stay `transparent`? I don't think "sdk error" is useful context.